test/e2e/framework.go

```diff
@@ -814,6 +814,14 @@ func (data *TestData) restartAntreaControllerPod(timeout time.Duration) (*corev1
 		if pod.Status.Phase != corev1.PodRunning || pod.DeletionTimestamp != nil {
 			return false, nil
 		}
+		for _, condition := range pod.Status.Conditions {
+			if condition.Type == corev1.PodReady {
+				if condition.Status == corev1.ConditionFalse {
+					return false, nil
+				}
+				break
+			}
+		}
```

Comment:
do you think the condition above is still needed now that we look at `pod.Status.Conditions`?

Reply:
Not needed if I add a check to ensure `corev1.PodReady` must exist as the `pod.Status.Conditions` is empty when the Pod is newly created. updated in latest patch.